libqtile/backend/wayland/qw/debug_util.c

```diff
@@ -0,0 +1,93 @@
+#include "debug_util.h"
+
+static void print_scene_node(struct wlr_scene_node *node, int depth, bool *draw_verticals, int child_i, int child_count, bool hide_rects) {
+    if (hide_rects && node->type == WLR_SCENE_NODE_RECT) {
+        return;
+    }
+
+    bool is_last = false;
+    if (child_i == child_count-1) {
+        is_last = true;
+    }
+
+    const char *type = "unknown";
+    switch (node->type) {
+        case WLR_SCENE_NODE_TREE:   type = "tree"; break;
+        case WLR_SCENE_NODE_RECT:   type = "rect"; break;
+        case WLR_SCENE_NODE_BUFFER: type = "buffer"; break;
+    }
+
+    const char* layer_name = "unknown";
+    switch (child_i) {
+        case LAYER_BACKGROUND:   layer_name = "LAYER_BACKGROUND"; break;
+        case LAYER_BOTTOM:       layer_name = "LAYER_BOTTOM"; break;
+        case LAYER_KEEPBELOW:    layer_name = "LAYER_KEEPBELOW"; break;
+        case LAYER_LAYOUT:       layer_name = "LAYER_LAYOUT"; break;
+        case LAYER_KEEPABOVE:    layer_name = "LAYER_KEEPABOVE"; break;
+        case LAYER_MAX:          layer_name = "LAYER_MAX"; break;
+        case LAYER_FULLSCREEN:   layer_name = "LAYER_FULLSCREEN"; break;
+        case LAYER_BRINGTOFRONT: layer_name = "LAYER_BRINGTOFRONT"; break;
+        case LAYER_TOP:          layer_name = "LAYER_TOP"; break;
+        case LAYER_OVERLAY:      layer_name = "LAYER_OVERLAY"; break;
+        case LAYER_END:          layer_name = "LAYER_END"; break;
+    }
+
+    struct qw_view *view = node->data;
+
+    // Print visual indent tree structure
+    char line[512] = {0};
```

Comment:
how is this length computed? having random hard-coded number is typically something you want to avoid in string buffer, because it may cause very bad effects